### PR TITLE
Create post favor flow

### PR DESCRIPTION
**Pull request by muneebazam**

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
+    private DatabaseWrapper() {
+        FirebaseFirestore.setLoggingEnabled(true);
+        FirebaseFirestoreSettings settings =
+                new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(true).build();
+        try {
+            firestore = FirebaseFirestore.getInstance();
+        } catch (Exception e) {
+            Log.d(TAG, e.toString());
+        }
+        firestore.setFirestoreSettings(settings);
```
Comment (arosu) [inline app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java]: If line 27 throws, then `firestore` is an un-initialized object.

Comment (muneebazam) [reply app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java]: you are right good sir

File: app/src/main/java/ch/epfl/favo/user/User.java [hunk 2]
```diff
@@ -53,17 +67,22 @@ public int getActiveAcceptingFavors() {
...
+  void setActiveAcceptingFavors(int activeAcceptingFavors) {
+    this.activeAcceptingFavors = activeAcceptingFavors;
+  }
+
   void setActiveRequestingFavors(int activeRequestingFavors) {
     this.activeRequestingFavors = activeRequestingFavors;
   }
+
+  void setNotificationId(String notificationId) {
+    this.notificationId =  notificationId;
```
Comment (mikanikos) [inline app/src/main/java/ch/epfl/favo/user/User.java]: Where do you set the notificationId?

File: app/src/main/java/ch/epfl/favo/user/User.java [hunk 3]
```diff
@@ -2,15 +2,19 @@
...
 public class User {
 
+  private String id;
   private String name;
   private String email;
   private String deviceId;
+  private String notificationId;
   private LocalDate birthDate;
   private int activeAcceptingFavors;
   private int activeRequestingFavors;
```
Comment (mikanikos) [inline app/src/main/java/ch/epfl/favo/user/User.java]: We also need a field for the Location of the user

File: app/src/main/java/ch/epfl/favo/user/User.java (hunk 2 above)
Comment (muneebazam) [reply app/src/main/java/ch/epfl/favo/user/User.java]: that is in the next PR, will be setting it in the SignInActivity.Java as we spoke about

File: app/src/main/java/ch/epfl/favo/user/User.java (hunk 3 above)
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/user/User.java]: I guess this is also part of the next PR

Comment (muneebazam) [reply app/src/main/java/ch/epfl/favo/user/User.java]: ok let me add that

File: app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java [hunk 4]
```diff
@@ -1,17 +1,59 @@
...
+            INSTANCE = new DatabaseWrapper();
+        }
+        return INSTANCE;
+    }
+
+    public static String generateRandomId() {
+
+        StringBuilder sb = new StringBuilder(ID_LENGTH);
+        for (int i = 0; i < ID_LENGTH; i++) {
+            int index = (int)(36 * Math.random());
```
Comment (samuelchassot) [inline app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java]: You should have used the ID_CHARS.length() (in a local var to keep performance)

File: app/src/main/java/ch/epfl/favo/favor/FavorUtil.java [hunk 5]
```diff
@@ -23,14 +30,22 @@ public static FavorUtil getSingleInstance() {
...
-   * @param description String containing 300 char (max) description of text.
-   * @param location Address or coordinates at which the favor is requested. TODO: post favor in DB
-   *     linked to user
+   * @param f A favor object.
    */
-  public void postFavor(String title, String description, String location) {
-
-    throw new NotImplementedException();
+  public void postFavor(Favor f) {
+    if (testUiMode) {return;} //will skip if testing the UI
```
Comment (samuelchassot) [inline app/src/main/java/ch/epfl/favo/favor/FavorUtil.java]: Bad idea to have test code in production codebase, you should find another way. (e.g. a dependency injection in the collectionWrapper)

**Pull request merged**